### PR TITLE
[develop] add ledger proof statement to blockchain state in GraphQL 

### PR DESCRIPTION
**Pull request by psteckler**
Added relevant commits from #14220 to allow merge-to-develop to succeed in CI there.

Had to make some other small changes (in 8692d18) because `develop` had other changes.

### CODE REVIEW

**Comment by psteckler** [comment]
!ci-build-me

File: src/lib/cli_lib/graphql_types.ml [hunk 1]
```diff
@@ -20,7 +20,7 @@ module Pending_snark_work = struct
     type t =
       { work_id : int
       ; fee_excess : Currency.Amount.Signed.t
-      ; supply_increase : Currency.Amount.t
+      ; supply_change : Currency.Amount.Signed.t
```
Comment (deepthiskumar) [inline src/lib/cli_lib/graphql_types.ml:23]: Same, why the renaming?

File: src/app/cli/src/init/client.ml [hunk 2]
```diff
@@ -1034,14 +1034,24 @@ let pending_snark_work =
...
                           { Cli_lib.Graphql_types.Pending_snark_work.Work
                             .work_id = w.work_id
                           ; fee_excess =
                               Currency.Amount.Signed.of_fee
-                                (to_signed_fee_exn f.sign
-                                   (Currency.Amount.to_fee f.fee_magnitude) )
-                          ; supply_increase = w.supply_change.fee_magnitude
+                                (to_signed_fee_exn fee_excess_left.sign
+                                   fee_excess_left.feeMagnitude )
+                          ; supply_change =
```
Comment (deepthiskumar) [inline src/app/cli/src/init/client.ml:1044]: Why is this renamed?

Comment (psteckler) [reply src/app/cli/src/init/client.ml:1044]: In GraphQL, `supplyIncrease` is deprecated in favor of `supplyChange`.

**Comment by psteckler** [comment]
!ci-build-me

**Review by deepthiskumar — APPROVED** [review]
